activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -1,32 +1,48 @@
 module ActiveSupport
   module Testing
     class SimpleStubs # :nodoc:
-      Stub = Struct.new(:object, :method_name, :original_method)
+      Stub = Struct.new(:object, :method_name, :original_method, :return_value)
```

Comment:
Think we could also get more mileage out of pushing some behavior into the Stub class.

Here's a spike:

```ruby

class Stub        
  attr_reader :return_values
  def initialize(object, method_name)
    @object, @method_name = object, method_name
    @new_name = "__simple_stub__#{method_name}"
    @return_values = []
  end
  def push(return_value)
    @return_values << return_value
    unless object.singleton_class.respond_to?(@new_name)
      stub = self
      object.singleton_class.send :alias_method, @new_name, @method_name
      object.define_singleton_method(method_name) { stub.return_values.last }
    end
  end
  def pop
    @return_values.pop
    if @return_values.empty?
      singleton_class = object.singleton_class
      singleton_class.send :undef_method, @method_name
      singleton_class.send :alias_method, @method_name, @new_name
      singleton_class.send :undef_method, @new_name
    end
  end
end
```

Which with (in SimpleStubs):

```ruby
def [](object, method_name)
  @stubs[[ object.object_id, method_name ]] ||= Stub.new(object, method_name)
end
```

Would be used like:

```ruby
simple_stubs[Date, :today].push now.to_date
```

I think capturing the return values in an array should let us avoid having to keep redefining the methods every stack push 😁